src/main_window.cpp

```diff
@@ -264,19 +264,51 @@ void MainWindow::installPkg()
     // Create game directory.
     auto gamesDirectory = readGamesDirectorySetting();
 
-    if (!QDir(gamesDirectory).mkdir(param.titleId())) {
-        QString msg(
-            "Cannot create directory %1 inside %2. "
-            "If you have a failed installation from a previous attempt, you will need to remove this directory before trying again.");
+    // Get Param information
+    auto appver = param.appver();
+    auto category = param.category();
+    auto shortContentId = param.shortContentId();
+    auto title = param.title();
+    auto titleId = param.titleId();
+
+    // Check if file is Patch/DLC.
+    bool patchOrDlc = false;
+    if (category.startsWith("gp") || category.contains("ac")) {
+        patchOrDlc = true;
+    }
+
+    // If PKG isn't a game, DLC, or Patch, don't allow.
+    if (!patchOrDlc && !category.startsWith("gd")) {
+            QString msg("PKG file is not a Patch, DLC, or a Game. Possibly a corrupted PKG?");
+
+            QMessageBox::critical(&progress, "Error (Not Game, Patch, or DLC)", msg);
```

Comment:
`Not Game, Patch, or DLC` is redundant because it is already in the message.

Reply:
Congrats, it's meant to be a simple summary if the user doesn't wanna read all that lol.

Reply:
I don't think it is a good idea. See https://ux.stackexchange.com/a/13701 for the reason.